deephaven/vscode-deephaven
# fix: Handle server disconnects / non-existing server

This PR improves the server connection experience / fixes some bugs:
* Status bar item now shows `Deephaven: Connecting...` while a connection is being initialized (before it immediately showed as connected)
* Status bar shows `Deephaven: Disconnected` if connection fails
* If client disconnects due to server stopping or anything else that results in an Jsapi ide 'disconnect' event, the status bar will be updated to reflect the status
* Updated status bar / quick pick menu icons

### Testing
**Setup**
If you clone this repo and run `npm install`, you should be able to `f5` to run the extension in debug mode.

**Test Steps**
* Start a DH server on `localhost:10000`
* Connect to server in extension. Should see status bar "Connecting..." then "Connected" status (the connecting status may happen too quickly to really see it).
* Kill server
* Should see status bar item switch to disconnected
* Attempt to re-connect to stopped server
* Should see "Connecting..." status for longer this time and then a "Failed to initialize Deephaven API" toast. Status should say "Disconnected"

fixes #4 fixes #5 fixes #6
